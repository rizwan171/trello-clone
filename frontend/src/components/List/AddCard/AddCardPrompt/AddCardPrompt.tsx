import { FiPlus } from "react-icons/fi";
import { AddCardPromptProps } from "../../../../types/components/AddCardPromptProps";

const AddCardPrompt = ({ setOpen }: AddCardPromptProps) => {

  return (
    <div className="flex flex-col transition-height ease-in-out duration-100 h-min">
      <div
        data-testid="add-card-prompt-container"
        className="flex hover:bg-trello-gray-500 rounded-ibsm items-center cursor-pointer text-trello-gray-600 p-1 mt-1"
        onClick={() => setOpen(true)}
      >
        <FiPlus size={20} />
        <h2 className="flex-1 ml-1.5">Add a card</h2>
      </div>
    </div>
  );
};

export default AddCardPrompt;
