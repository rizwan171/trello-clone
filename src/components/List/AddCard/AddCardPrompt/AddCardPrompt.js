import React from "react";
import { FiPlus } from "react-icons/fi";

const AddCardPrompt = ({ setOpen }) => {
  return (
    <div
      className="flex hover:bg-trello-gray-500 rounded-ibsm items-center cursor-pointer text-trello-gray-600 p-1"
      onClick={() => setOpen(true)}
    >
      <FiPlus size={20} />
      <h2 className="flex-1 ml-1.5">Add a card</h2>
    </div>
  );
};

export default AddCardPrompt;
