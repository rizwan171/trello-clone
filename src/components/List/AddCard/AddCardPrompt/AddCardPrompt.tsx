import { FiPlus } from "react-icons/fi";
import { AddCardPromptProps } from "../../../../types/components/AddCardPromptProps";

const AddCardPrompt = ({ open, setOpen }: AddCardPromptProps): JSX.Element => {
  const inlineStyle = open ? { height: "min-content" } : { height: 0, overflow: "hidden" };

  return (
    <div className="flex flex-col transition-height ease-in-out duration-100" style={inlineStyle}>
      <div
        className="flex hover:bg-trello-gray-500 rounded-ibsm items-center cursor-pointer text-trello-gray-600 p-1 mt-1"
        onClick={() => setOpen(true)}
      >
        <FiPlus size={20} />
        <h2 className="flex-1 ml-1.5">Add a card</h2>
      </div>
    </div>
  );
};

export default AddCardPrompt;
