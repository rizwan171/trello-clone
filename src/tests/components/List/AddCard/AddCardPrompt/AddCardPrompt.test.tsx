import { fireEvent, render, screen } from "@testing-library/react";
import AddCardPrompt from "../../../../../components/List/AddCard/AddCardPrompt/AddCardPrompt";

describe("AddCardPrompt", () => {
  let mockSetOpen: jest.Mock;

  beforeEach(() => {
    mockSetOpen = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render successfully when open", () => {
    const view = render(<AddCardPrompt open={true} setOpen={mockSetOpen} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it("should render successfully when closed", () => {
    const view = render(<AddCardPrompt open={false} setOpen={mockSetOpen} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it("should open the add card form", () => {
    render(<AddCardPrompt open={true} setOpen={mockSetOpen} />);

    const addCardPromptContainer = screen.getByTestId("add-card-prompt-container") as HTMLDivElement;
    expect(addCardPromptContainer).toBeInTheDocument();

    fireEvent.click(addCardPromptContainer);

    expect(mockSetOpen).toHaveBeenCalledWith(true);
  });
});
