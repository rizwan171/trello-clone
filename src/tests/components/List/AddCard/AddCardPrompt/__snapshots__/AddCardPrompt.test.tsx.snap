// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardPrompt should render successfully when closed 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col transition-height ease-in-out duration-100"
    style="height: 0px; overflow: hidden;"
  >
    <div
      class="flex hover:bg-trello-gray-500 rounded-ibsm items-center cursor-pointer text-trello-gray-600 p-1 mt-1"
      data-testid="add-card-prompt-container"
    >
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
      </svg>
      <h2
        class="flex-1 ml-1.5"
      >
        Add a card
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AddCardPrompt should render successfully when open 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col transition-height ease-in-out duration-100"
  >
    <div
      class="flex hover:bg-trello-gray-500 rounded-ibsm items-center cursor-pointer text-trello-gray-600 p-1 mt-1"
      data-testid="add-card-prompt-container"
    >
      <svg
        fill="none"
        height="20"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
      </svg>
      <h2
        class="flex-1 ml-1.5"
      >
        Add a card
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
